Tests/Integration/LoRaDataRequestHandlerConnectionAffinityIntegrationTests.cs

```diff
@@ -64,55 +64,89 @@ public LoRaDataRequestHandlerConnectionAffinityIntegrationTests(ITestOutputHelpe
         }
 
         [Theory]
-        [InlineData(true)]
-        [InlineData(false)]
-        [InlineData(null)]
-        public async Task When_Gateway_Does_Not_Own_Connection_It_Should_Delay(bool? connectionOwner)
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(false, false)]
+        [InlineData(null, true)]
+        public async Task When_Gateway_Does_Not_Own_Connection_It_Should_Delay_If_Delay_Enabled(bool? connectionOwner, bool processingDelayEnabled)
         {
             // arrange
             this.deviceMock.Object.IsConnectionOwner = connectionOwner;
 
+            if (!processingDelayEnabled)
+                ServerConfiguration.ProcessingDelayInMilliseconds = 0;
```

Comment:
[nit] Here we are setting the configuration in order to ensure that `IsProcessingDelayEnabled()` will return the right result. However if we change the implementation (for example so that 0 is considered as delay enabled) then the tests would need to be fixed. Since we have the `dataRequestHandlerMock` shouldn't we maybe use it here as well to mock out the implementation of the `IsProcessingDelayEnabled()`?

Reply:
Makes sense, done in 5ff5499bc0c9f0c480246e663cbed29d0c4fe500